digitalocean/droplet/resource_droplet_test.go

```diff
@@ -477,6 +477,31 @@ func TestAccDigitalOceanDroplet_EnableAndDisableBackups(t *testing.T) {
 	})
 }
 
+func TestAccDigitalOceanDroplet_ChangeBackupPolicy(t *testing.T) {
```

Comment:
I think there are some additional scenarios we should test. 

- [x] Create a Droplet with a backup policy 
- [ ] Disable backups on a Droplet with a backup policy 
- [ ] Enable backups with a policy for a Droplet initially created without on
- [ ] Update the backup policy for a Droplet

Reply:
Should be good now.

Reply:
Should be good now.

Reply:
It would be great to write some additional tests covering these. You can have multiple steps in a single test. Something like:

```diff
diff --git a/digitalocean/droplet/resource_droplet_test.go b/digitalocean/droplet/resource_droplet_test.go
index 80130ca2..c943aa19 100644
--- a/digitalocean/droplet/resource_droplet_test.go
+++ b/digitalocean/droplet/resource_droplet_test.go
@@ -480,6 +480,17 @@ func TestAccDigitalOceanDroplet_EnableAndDisableBackups(t *testing.T) {
 func TestAccDigitalOceanDroplet_ChangeBackupPolicy(t *testing.T) {
 	var droplet godo.Droplet
 	name := acceptance.RandomTestName()
+	backupPolicy := `  backup_policy {
+    plan    = "daily"
+    hour    = 4
+  }
+`
+	updatedPolicy := `  backup_policy {
+    plan    = "weekly"
+    weekday = "MON"
+    hour    = 0
+  }
+`
 
 	resource.ParallelTest(t, resource.TestCase{
 		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
@@ -487,7 +498,56 @@ func TestAccDigitalOceanDroplet_ChangeBackupPolicy(t *testing.T) {
 		CheckDestroy:      acceptance.TestAccCheckDigitalOceanDropletDestroy,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccCheckDigitalOceanDropletConfig_ChangeBackupPolicy(name),
+				Config: testAccCheckDigitalOceanDropletConfig_ChangeBackupPolicy(name, ""),
+				Check: resource.ComposeTestCheckFunc(
+					acceptance.TestAccCheckDigitalOceanDropletExists("digitalocean_droplet.foobar", &droplet),
+					resource.TestCheckResourceAttr(
+						"digitalocean_droplet.foobar", "backups", "true"),
+				),
+			},
+			{
+				Config: testAccCheckDigitalOceanDropletConfig_ChangeBackupPolicy(name, backupPolicy),
+				Check: resource.ComposeTestCheckFunc(
+					acceptance.TestAccCheckDigitalOceanDropletExists("digitalocean_droplet.foobar", &droplet),
+					resource.TestCheckResourceAttr(
+						"digitalocean_droplet.foobar", "backup_policy.0.plan", "daily"),
+					resource.TestCheckResourceAttr(
+						"digitalocean_droplet.foobar", "backup_policy.0.hour", "4"),
+				),
+			},
+			{
+				Config: testAccCheckDigitalOceanDropletConfig_ChangeBackupPolicy(name, updatedPolicy),
+				Check: resource.ComposeTestCheckFunc(
+					acceptance.TestAccCheckDigitalOceanDropletExists("digitalocean_droplet.foobar", &droplet),
+					resource.TestCheckResourceAttr(
+						"digitalocean_droplet.foobar", "backup_policy.0.plan", "weekly"),
+					resource.TestCheckResourceAttr(
+						"digitalocean_droplet.foobar", "backup_policy.0.weekday", "MON"),
+					resource.TestCheckResourceAttr(
+						"digitalocean_droplet.foobar", "backup_policy.0.hour", "0"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDigitalOceanDroplet_WithBackupPolicy(t *testing.T) {
+	var droplet godo.Droplet
+	name := acceptance.RandomTestName()
+	backupPolicy := `  backup_policy {
+    plan    = "weekly"
+    weekday = "MON"
+    hour    = 0
+  }
+`
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      acceptance.TestAccCheckDigitalOceanDropletDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckDigitalOceanDropletConfig_ChangeBackupPolicy(name, backupPolicy),
 				Check: resource.ComposeTestCheckFunc(
 					acceptance.TestAccCheckDigitalOceanDropletExists("digitalocean_droplet.foobar", &droplet),
 					resource.TestCheckResourceAttr(
@@ -1053,7 +1113,7 @@ resource "digitalocean_droplet" "foobar" {
 }`, name, defaultSize, defaultImage)
 }
 
-func testAccCheckDigitalOceanDropletConfig_ChangeBackupPolicy(name string) string {
+func testAccCheckDigitalOceanDropletConfig_ChangeBackupPolicy(name, backupPolicy string) string {
 	return fmt.Sprintf(`
 resource "digitalocean_droplet" "foobar" {
   name      = "%s"
@@ -1062,12 +1122,8 @@ resource "digitalocean_droplet" "foobar" {
   region    = "nyc3"
   user_data = "foobar"
   backups   = true
-  backup_policy {
-    plan    = "weekly"
-    weekday = "MON"
-    hour    = 0
-  }
-}`, name, defaultSize, defaultImage)
+%s
+}`, name, defaultSize, defaultImage, backupPolicy)
 }
 
 func testAccCheckDigitalOceanDropletConfig_DropletAgent(keyName, testAccValidPublicKey, dropletName, image, agent string) string {
```

Reply:
Added in [087cda0](https://github.com/digitalocean/terraform-provider-digitalocean/pull/1261/commits/087cda029f9bc9938613e5512f18b782e3c83fe1)

Reply:
Added one more test to ensure that correct policy applied instead of default policy after backups were re-enabled: [924c3ed](https://github.com/digitalocean/terraform-provider-digitalocean/pull/1261/commits/924c3edf4ad592be9842bf5c38fe80d3a8236f21)